### PR TITLE
feat: AT4-44 added backend to display data from database according to…

### PR DESCRIPTION
**Pull request by AraojoBenedict**
- added session
- added backend to display data from the database according to position
- functional filter button to display the chosen position
- added sorting from highest to lowest on each position
- fixed style path
- commented the script that add candidate card

### CODE REVIEW

**Comment by Carl-Tabuso** [comment]
Your branch is outdated. Kindly rebase it to fetch the latest changes from base branch development.

**Comment by AraojoBenedict** [comment]
Successfully rebased and updated refs/heads/AT4-44.

**Comment by Carl-Tabuso** [comment]
Earlier, I was talking about your head branch being outdated. I am requesting you once again to rebase your head branch from the development branch before we proceed with the merging.

image

**Review by Carl-Tabuso — CHANGES_REQUESTED** [review]
Your code is quite hard to understand because some are not indented properly. There are also too much linebreaks from one line of code to another. See images below.

image
image
image

Your SQL queries are also of procedural types. In OOP PHP, we are mostly using the arrow notation (->) instead of the dot (.) one. Let's be consistent with how we work on the server-side and make sure we adhere to what has been decided by the development team.

image
image
image

May i also request for an SQL file i can import in phpmyadmin so i can further explore and test your code before we get into merging.

**Comment by AraojoBenedict** [comment]
[SQL File.txt](https://github.com/BSIT-3-1-APPDEV/PUPSRC-AutomatedElectionSystem/files/15077755/SQL.File.txt)

File: src/result-generation.php [hunk 1]
```diff
@@ -2,357 +2,225 @@
...
                                             </div>
-                                            <p><?php echo $vote_count; ?> Votes</p>
                                         </div>
                                     </div>
                                 </div>
-                            </div>
-                        <?php } ?>
-                    <?php
-                    } else {      
+     <?php
```
Comment (Carl-Tabuso) [inline src/result-generation.php:187]: properly indent this next time.

File: src/result-generation.php [hunk 2]
```diff
@@ -2,357 +2,225 @@
...
+            $stmt->bind_param("s", $position);
+            $stmt->execute();
+            $result = $stmt->get_result();
+
+            // Check if candidates exist for the position
+            if ($result->num_rows > 0) {
+                // Loop through candidates
+                while ($candidate_data = $result->fetch_assoc()) {
+                    $candidate_id = $candidate_data['candidate_id'];
+                    $vote_count_query = "SELECT COUNT(*) as vote_count FROM vote WHERE candidate_id = $candidate_id";
```
Comment (Carl-Tabuso) [inline src/result-generation.php:160]: Use prepared sql statements next time. This is to avoid possible SQL injections to you r code.

**Review by Carl-Tabuso — APPROVED** [review]
See the comments and make sure to address them as fix type next time.

**Pull request merged**